python/hsfs/core/job_scheduler.py

```diff
@@ -14,25 +14,34 @@
 #   limitations under the License.
 #
 import pandas as pd
-from datetime import datetime
-from typing import Optional, Union
+from datetime import datetime, timedelta, date
+from typing import Any, Dict, Optional, Union
 import humps
 import json
+import re
 
 from hsfs import util
+from hsfs.core import job_scheduler_engine
 
 
 class JobScheduler:
     def __init__(
         self,
-        start_date_time: Union[int, str, datetime, pd.Timestamp],
-        job_frequency: str,
+        start_date_time: Union[int, str, datetime, date, pd.Timestamp],
+        job_frequency: Optional[str] = "CUSTOM",
+        cron_expression: Optional[str] = None,
```

Comment:
To keep it simpler, have you thought of merging these two parameters?
Something like job_frequency with possible values "HOURLY", "DAILY", "WEEKLY" or "/cron expression/".

Reply:
Yes I thought about it. I think it is something you and the rest of the team will have to decide when finalising docs and tutorial. I definitely agree think it is easier UX. I can fix it now if you think it is a no-brainer